Sweetener.Logging/LogPattern.cs

```diff
@@ -0,0 +1,385 @@
+                    // There are currently no patterns with numbers enabled
+                    if (char.IsLetter(c) && nameEnd == -1)
+                    {
+                        if (nameStart == -1)
+                            nameStart = i;
+                    }
+                    else if (c == ' ')
+                    {
+                        if (nameStart != -1 && nameEnd == -1)
+                            nameEnd = i;
+                    }
+                    else if (c == ',' || c == ':' || c == '}')
+                    {
+                        if (nameEnd == -1)
+                            nameEnd = i;
+
+                        break;
+                    }
+                    else
+                    {
+                        throw new FormatException("Input string was not in a correct format");
+                    }
+                }
+
+                // Did we reach the end without breaking?
+                if (i >= pattern.Length)
+                    throw new FormatException("Input string was not in a correct format");
+
+                string name = pattern.Substring(nameStart, nameEnd - nameStart);
+
+                // TODO: Case insensitive?
+                switch (name)
+                {
+                    case "msg":
+                    case "message":
+                        return Parameter.Message; 
+                    case "ts":
+                    case "timestamp":
+                        return Parameter.Timestamp; 
+                    case "l":
+                    case "level":
+                        return Parameter.Level;
+                    case "pid":
+                    case "processId":
+                        return Parameter.ProcessId;
+                    case "pn":
+                    case "processName":
+                        return Parameter.ProcessName;
+                    case "tid":
+                    case "threadId":
+                        return Parameter.ThreadId;
+                    case "tn":
+                    case "threadName":
+                        return Parameter.ThreadName;
+                    default:
+                        throw new FormatException($"Unrecognized pattern variable '{name}'.");
+                }
+            }
+        }
+
+        /// <summary>
+        /// A known variable in the log layout pattern.
+        /// </summary>
+        public enum Parameter
+        {
+            /// <summary>
+            /// The log message.
+            /// </summary>
+            Message = 0,
+
+            /// <summary>
+            /// The timestamp when the log request was made.
+            /// </summary>
+            /// <remarks>
+            /// By default, the timestamp is UTC.
+            /// </remarks>
+            Timestamp = 1,
+
+            /// <summary>
+            /// The <see cref="LogLevel"/> associated with the log entry.
+            /// </summary>
+            Level = 2,
```

Comment:
Level is currently treated as an `enum`. Should it be a `string` instead?

Reply:
What's the advantage of treating it as a string? If you ever need a string can't you just toString() the enum?

Reply:
The only advantage I see is that when a user includes “{level}” in their pattern, by default the value is “Warn,” “Fatal,” etc instead of 3, 4, etc